flink-dstl/flink-dstl-dfs/src/main/java/org/apache/flink/changelog/fs/BatchingStateChangeUploader.java

```diff
@@ -206,16 +251,27 @@ private synchronized void setErrorSafe(Throwable t) {
     }
 
     private static UploadTask wrapWithSizeUpdate(
-            UploadTask uploadTask, long preComputedTaskSize, LongAdder inflightSize) {
+            UploadTask uploadTask, Runnable releaseCapacityCallback) {
         return new UploadTask(
                 uploadTask.changeSets,
                 result -> {
-                    inflightSize.add(-preComputedTaskSize);
-                    uploadTask.successCallback.accept(result);
+                    try {
+                        releaseCapacityCallback.run();
```

Comment:
Why indirect `Runnable`? Can not we call `releaseCapacity(size)` directly here?

Reply:
My intention was to make it clear what is captured in terms of references at runtime;
but apparrently it doesn't make it more clear, so I'll change back to a direct call and make `wrapWithSizeUpdate` non-static.